pytorch_toolkit/ote/ote/modules/arg_parsers/default.py

```diff
@@ -14,27 +14,102 @@
  limitations under the License.
 """
 
-from ote.api import (train_args_parser,
-                     test_args_parser,
-                     export_args_parser,
-                     compression_args_parser)
+import argparse
+import yaml
 
 from ..registry import ARG_PARSERS
 
 
 @ARG_PARSERS.register_module()
 class DefaultArgParser:
-    def __init__(self):
-        super(DefaultArgParser, self).__init__()
+    def _get_base_parser_for_train_compression(self, model_template_path):
+        parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter)
+        with open(model_template_path, 'r') as f:
+            model_template = yaml.safe_load(f)
 
-    def get_train_parser(self, config_path):
-        return train_args_parser(config_path)
+        parser.add_argument('--train-ann-files', required=True,
+                            help='Comma-separated paths to training annotation files.')
+        parser.add_argument('--train-data-roots', required=True,
+                            help='Comma-separated paths to training data folders.')
+        parser.add_argument('--val-ann-files', required=True,
+                            help='Comma-separated paths to validation annotation files.')
+        parser.add_argument('--val-data-roots', required=True,
+                            help='Comma-separated paths to validation data folders.')
+        parser.add_argument('--resume-from', default='',
+                            help='Resume training from previously saved checkpoint')
+        parser.add_argument('--load-weights', default='',
+                            help='Load only weights from previously saved checkpoint')
+        parser.add_argument('--save-checkpoints-to', default='/tmp/checkpoints',
+                            help='Location where checkpoints will be stored')
+        parser.add_argument('--batch-size', type=int,
+                            default=model_template['hyper_parameters']['basic']['batch_size'],
+                            help='Size of a single batch during training per GPU.')
+        parser.add_argument('--gpu-num', type=int,
+                            default=model_template['gpu_num'],
+                            help='Number of GPUs that will be used in training, 0 is for CPU mode.')
+        parser.add_argument('--tensorboard-dir',
+                            help='Location where tensorboard logs will be stored.')
+        parser.add_argument('--config', default=model_template['config'], help=argparse.SUPPRESS)
 
-    def get_test_parser(self, config_path):
-        return test_args_parser(config_path)
+        return parser
 
-    def get_export_parser(self, config_path):
-        return export_args_parser(config_path)
+    def get_train_parser(self, model_template_path):
```

Comment:
I'm not sure indeed 🙂 
As far as I can see, `api.py` is not required indeed, since all the classes are derived from one base class.
So, I removed `api.py`

But if you suppose that it should be returned back, I'm ok with that.
@Ilya-Krylov , what is your opinion?

Reply:
it is violable